### PR TITLE
[BUGFIX] Allow to search for nodes without a Document parent

### PR DESCRIPTION
**Pull request by kdambekalns**
This checks whether a documentNode has been found before rendering a
link to the frontend for nodes that have been found by the nodes
service controller.

Fixes NEOS-1695

### CODE REVIEW

**Comment by kdambekalns** [comment]
The question is: do we need to output any "explanation" if no frontend URI can be rendered (as in the change), or not?

**Comment by robertlemke** [comment]
I guess an explanation is okay for the time being, just an empty URL may be confusing. But maybe rather write something like "This node can't be access through a public URL"?

**Comment by kdambekalns** [comment]
Adjusted the explanation (it is visible in the editor suggestions, so it should be user-friendly) and fixed the template for showing node details.

File: TYPO3.Neos/Resources/Private/Templates/Service/Nodes/Index.html [hunk 1]
```diff
@@ -10,11 +10,20 @@
...
+							<f:then>
+								<f:alias map="{relativeUrl: '{neos:uri.node(node: documentNode, absolute: false, resolveShortcuts: false)}'}">
+									<a href="{neos:uri.node(node: documentNode, absolute: true, resolveShortcuts:false)}" class="node-frontend-uri">
+										{f:if(condition: relativeUrl, then: '{relativeUrl}', else: '{node.path}')}
+									</a>
+								</f:alias>
+							</f:then>
+							<f:else>
+								<a class="node-frontend-uri">
+									This node can't be accessed through a public URL
```
Comment (kitsunet) [inline TYPO3.Neos/Resources/Private/Templates/Service/Nodes/Index.html]: Should be translatable

File: TYPO3.Neos/Resources/Private/Templates/Service/Nodes/Show.html [hunk 2]
```diff
@@ -18,9 +18,18 @@
...
-					</neos:link.node>
+					<f:if condition="{documentNode}">
+						<f:then>
+							<neos:link.node node="{documentNode}" absolute="true" resolveShortcuts="false" class="node-frontend-uri">
+								<neos:uri.node node="{documentNode}" absolute="false" resolveShortcuts="false" />
+							</neos:link.node>
+						</f:then>
+						<f:else>
+							<a class="node-frontend-uri">
+								This node can't be accessed through a public URL
```
Comment (kitsunet) [inline TYPO3.Neos/Resources/Private/Templates/Service/Nodes/Show.html]: Same here

**Comment by kdambekalns** [comment]
Translation is now possible, added some missing XLIFF entries along the way.

**Comment by kdambekalns** [comment]
:+1: from my side

**Comment by kitsunet** [comment]
:+1:

**Comment by aertmann** [comment]
any reason this wasn't targeted for 2.0?

**Comment by kdambekalns** [comment]
AFAIK searching outside the current site is only possible on master, thus the problem doesn't exist on 2.0. Wrong?

**Comment by aertmann** [comment]
don't know, but if so all fine then..

however the translation stuff should go to 2.0

**Comment by kdambekalns** [comment]
Well, the translation stuff, as noted, is invisible. The output is consumed by our JS on the client, never by humans. I was close to removing the translate calls, but… The new one is the only that will actually end up being visible to editors.

**Comment by aertmann** [comment]
okay thanks for explaining

**Pull request merged**